gemfire-core/src/test/java/com/gemstone/gemfire/internal/offheap/GemFireChunkFactoryJUnitTest.java

```diff
@@ -0,0 +1,138 @@
+ * limitations under the License.
+ */
+
+package com.gemstone.gemfire.internal.offheap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import com.gemstone.gemfire.LogWriter;
+import com.gemstone.gemfire.internal.cache.EntryEventImpl;
+import com.gemstone.gemfire.test.junit.categories.UnitTest;
+
+@Category(UnitTest.class)
+public class GemFireChunkFactoryJUnitTest {
+
+  private MemoryAllocator ma;
+
+  @Before
+  public void setUp() {
+    OutOfOffHeapMemoryListener ooohml = mock(OutOfOffHeapMemoryListener.class);
+    OffHeapMemoryStats stats = mock(OffHeapMemoryStats.class);
+    LogWriter lw = mock(LogWriter.class);
+
+    ma = SimpleMemoryAllocatorImpl.create(ooohml, stats, lw, 1, OffHeapStorage.MIN_SLAB_SIZE * 1, OffHeapStorage.MIN_SLAB_SIZE);
+  }
+
+  @After
+  public void tearDown() {
+    SimpleMemoryAllocatorImpl.freeOffHeapMemory();
+  }
+
+  private GemFireChunk createChunk(Object value) {
+    byte[] v = EntryEventImpl.serialize(value);
+
+    boolean isSerialized = true;
+    boolean isCompressed = false;
+
+    GemFireChunk chunk = (GemFireChunk) ma.allocateAndInitialize(v, isSerialized, isCompressed, GemFireChunk.TYPE);
+    chunk.setSerializedValue(v);
+    chunk.setCompressed(isCompressed);
+    chunk.setSerialized(isSerialized);
+
+    return chunk;
+  }
+
+  @Test
+  public void factoryShouldCreateNewChunkWithGivenAddress() {
+    GemFireChunk chunk = createChunk(Long.MAX_VALUE);
+
+    ChunkFactory factory = new GemFireChunkFactory();
+    Chunk newChunk = factory.newChunk(chunk.getMemoryAddress());
+
+    assertNotNull(newChunk);
+    assertEquals(GemFireChunk.class, newChunk.getClass());
+
+    assertThat(newChunk.getMemoryAddress()).isEqualTo(chunk.getMemoryAddress());
+
+    chunk.release();
+  }
+
+  @Test
+  public void factoryShouldCreateNewChunkWithGivenAddressAndType() {
+    GemFireChunk chunk = createChunk(Long.MAX_VALUE);
+
+    ChunkFactory factory = new GemFireChunkFactory();
+    Chunk newChunk = factory.newChunk(chunk.getMemoryAddress(), GemFireChunk.TYPE);
+
+    assertNotNull(newChunk);
+    assertEquals(GemFireChunk.class, newChunk.getClass());
+
+    assertThat(newChunk.getMemoryAddress()).isEqualTo(chunk.getMemoryAddress());
+    assertThat(newChunk.getChunkType()).isEqualTo(GemFireChunk.TYPE);
+
+    chunk.release();
+  }
+
+  @Test
+  public void shouldGetChunkTypeFromAddress() {
+    byte[] v = EntryEventImpl.serialize(Long.MAX_VALUE);
+
+    boolean isSerialized = true;
+    boolean isCompressed = false;
+
+    GemFireChunk chunk = (GemFireChunk) ma.allocateAndInitialize(v, isSerialized, isCompressed, GemFireChunk.TYPE);
+    chunk.setSerializedValue(v);
```

Comment:
allocateAndInitialize call setSerializeValue, setSerialized, and setCompressed. So the test does not need to call them.